docs/content/preview/architecture/transactions/isolation-levels.md

```diff
@@ -67,11 +67,49 @@ The following matrix shows conflicts between locks of different types at a high
+Although described here as a separate lock type for simplicity, the snapshot isolation write lock type is actually implemented internally as a combination of the other two lock types.  That is, taking a single snapshot isolation write lock is equivalent to taking both a serializable read lock and a serializable write lock.
+
+## Locking granularities
+
+Locks can be taken at many levels of granularity.  For example, a serializable read lock could be taken at the level of an entire tablet, a single row, or a single column of a single row.  Such a lock will block attempts to take write locks at that or lower granularities. Thus, for example, a read lock taken at the row level will block attempts to write to that entire row or any sub column of it.
+
+In addition to the above-mentioned levels of granularity, locks in DocDB can be taken at prefixes of the primary key, treating the hash columns as a single unit.  For example, if we have created a YSQL table via:
+```sql
+CREATE TABLE test (h1 INT, h2 INT, r1 INT, r2 INT, v INT w INT PRIMARY KEY ((h1,h2) HASH, r1 ASC, r2 ASC);
+```
+
+then we could lock at any of the following:
+- the entire tablet
+- all rows having h1=2, h2=3
+- all rows having h1=2, h2=3, r1=4
+- the row having h1=2, h2=3, r1=4, r2=5
+- column v of the row having h1=2, h2=3, r1=4, r2=5
+
+With YCQL, granularities exist below the column level; for example only one key of a column map can be locked.
+
+## Efficiently detecting conflicts between locks of different granularities
+
+The straightforward way to handle locks of different granularities would be to have a map from location to lock types.  However, this is too inefficient for detecting conflicts: attempting, for example, to add a lock at the tablet level would require checking for locks at every row and column in that tablet.
+
+To make conflict detection efficient, YugabyteDB stores extra information at each location about the locks at lower granularities below it.  In particular, it takes a normal lock at the original granularity and weaker versions of that lock at all the granularities that enclose the original granularity.  We call the normal locks _strong_ locks and the weaker variants _weak_ locks.
+
+As an example, pretend we have only tablet- and row-level granularities. To take a serializable write lock at the row level, we would take a strong write lock at the row level and a weak write lock at the tablet level.  To take a serializable read lock at the tablet level, we would just take a strong read lock at the tablet level.
```

Comment:
```suggestion
As an example, suppose you have only tablet- and row-level granularities. To take a serializable write lock at the row level, you would take a strong write lock at the row level and a weak write lock at the tablet level.  To take a serializable read lock at the tablet level, you would just take a strong read lock at the tablet level.
```

Reply:
The current version reads better to me because the connotations are that  we are temporarily changing the system.  Somehow I don't feel "you" has permission to do that.

Reply:
What's really doing the work here - the database? As noted, 'we' is typically avoided, but I'm not that fussed about it.